app/decorators/backend_api_decorator.rb

```diff
@@ -18,4 +18,18 @@ def as_json(options = {})
   def backend_api?
     true
   end
+
+  def links
+    [
+      { name: 'Edit', path: h.edit_provider_admin_backend_api_path(object) },
+      { name: 'Overview', path: h.provider_admin_backend_api_path(object) },
+      { name: 'Analytics', path: h.provider_admin_backend_api_stats_usage_path(object) },
+      { name: 'Methods & Metrics', path: h.provider_admin_backend_api_metrics_path(object) },
+      { name: 'Mapping Rules', path: h.provider_admin_backend_api_mapping_rules_path(object) },
+    ]
+  end
+
+  def products_count
+    object.services.size
```

Comment:
I don't know how this is done internally, but if it does the following, it is scary as hell 😱
1. backend_api left outer join backend_api_config left outer join services
2. count distinct services.id where backend_api.id is this one

instead, we should do this:
1. backend_api left outer join backend_api_config
2. count distinct backend_api_config.service_id where backend_api.id is this one

Reply:
What about calling `service.backend_api_configs` like it's done in `Account > Backends`?
https://github.com/3scale/porta/blob/ed7ad0d57ab2c8de7d4241afd5008f89de31fc59/app/controllers/api/backend_usages_controller.rb#L16

Reply:
I see what you're suggesting and indeed it would be an improvement:

```ruby
object = BackendApi.find(2)

object.services.count
# SELECT COUNT(*) FROM `services` INNER JOIN `backend_api_configs` ON `services`.`id` = `backend_api_configs`.`service_id` WHERE `backend_api_configs`.`backend_api_id` = 2

object.backend_api_configs.select(:service_id).distinct.count
# SELECT DISTINCT COUNT(DISTINCT `backend_api_configs`.`service_id`) FROM `backend_api_configs` WHERE `backend_api_configs`.`backend_api_id` = 2
```

However, I wonder if it's worth the effort. The proper way of optimising this would be by caching this count, but we don't have the time to do this now. Perhaps we can move on with this the way it is and consider another round of improvements later, with caching and all?

Reply:
Yep, makes sense all of it! 💪  👍